frontend/src/firebase/firebaseInteractor.ts

```diff
@@ -468,6 +468,13 @@ export default class FirebaseInteractor {
     userId: string,
     sessionId: number
   ): Promise<SessionStats> {
+    let user = await this.getUser(userId);
+    if (user == null) {
+      throw new Error(
+        `Requested user ${userId} session dashboard does not exist`
+      );
+    }
```

Comment:
I don't think we need this here. The dashboard already has the requested user, can we pass it to the SessionDashboard through there?

Reply:
we tried this, but the name would show up as `undefined` if we didn't get to the page through the user's dashboard (like if a researcher were to just visit the link directly rather than click through the user page), since`requestedStudent` would never be set in that case.

Reply:
Gotcha. I'm not sure if this is better than figuring out why requested student turns up as `undefined`. I am good with it, but do wonder why it does not get fetched correctly :shrug:

Reply:
i'm pretty sure it's because we check for `params.id` here https://github.com/sandboxnu/vocab-buddy/blob/master/frontend/src/pages/Dashboard/Dashboard.tsx#L534. I could change it to also load the requested user on a session page there but either way we still have to fetch the user

Reply:
Yeah I think changing it to use the existing behavior through the dashboard might be better, so that when we go back to the dashboard the data is already loaded. If you could update it that would be awesome.

Reply:
updated!